vncviewer/GestureHandler.cxx

```diff
@@ -201,9 +180,16 @@ bool GestureHandler::hasDetectedGesture() {
 void GestureHandler::trackTouch(const XIDeviceEvent *ev) {
   GHTouch ght;
 
+  gettimeofday(&ght.started, NULL);
   ght.last_x = ght.first_x = ev->event_x;
   ght.last_y = ght.first_y = ev->event_y;
 
+  // Did it take too long between touches that we should no longer
+  // consider this a single gesture?
+  if ((tracked.size() > 0) &&
+      (rfb::msSince(&tracked.begin()->second.started) > GH_MULTITOUCH_TIMEOUT))
+    return;
```

Comment:
Shouldn't we set an invalid state here? Otherwise a slow two finger gesture will be interpreted as a single finger one, which will most likely be confusing.

Reply:
TBH I don't really agree that the longpress timeout and gesture timeout need to be separate. IMO it shouldn't matter if it takes 3 hours for the second finger to come down, so long as nothing else (e.g. a longpress timeout, a release, or a movement) has triggered a state in the mean time. This was the idea originally.

Anyhow, see e9366e21